src/clusterfuzz/_internal/base/utils.py

```diff
@@ -673,16 +675,34 @@ def restart_machine():
     os.system('sudo shutdown -r now')
 
 
-def search_bytes_in_file(search_bytes, file_handle):
+def search_bytes_in_file(search_bytes,
+                         file_handle,
+                         buffer_length=DEFAULT_SEARCH_BUFFER_LENGTH):
   """Helper to search for bytes in a large binary file without memory
   issues.
   """
-  # TODO(aarya): This is too brittle and will fail if we have a very large
-  # line.
-  for line in file_handle:
-    if search_bytes in line:
+  assert buffer_length >= len(search_bytes)
+  buf = bytearray(buffer_length)
+  offset = 0
+  length = len(search_bytes) - 1
+  # We read right after the previously copied offset. This ensures we're
+  # perfectly matching the byte sequence above.
+  while read_len := file_handle.readinto(memoryview(buf)[offset:]):
```

Comment:
Let's immediately wrap `buf` in a `memoryview`. I think things will still work, and it will ensure we never copy `buf` accidentally?